src/Sarif/NotYetAutoGenerated/Location.cs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.CodeDom.Compiler;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// A location within a programming artifact.
+    /// </summary>
+    [DataContract]
+    [GeneratedCode("Microsoft.Json.Schema.ToDotNet", "0.58.0.0")]
+    public partial class Location : PropertyBagHolder, ISarifNode
+    {
+        public static IEqualityComparer<Location> ValueComparer => LocationEqualityComparer.Instance;
+
+        public bool ValueEquals(Location other) => ValueComparer.Equals(this, other);
+        public int ValueGetHashCode() => ValueComparer.GetHashCode(this);
+
+        /// <summary>
+        /// Gets a value indicating the type of object implementing <see cref="ISarifNode" />.
+        /// </summary>
+        public SarifNodeKind SarifNodeKind
+        {
+            get
+            {
+                return SarifNodeKind.Location;
+            }
+        }
+
+        /// <summary>
+        /// Identifies the file and region.
+        /// </summary>
+        [DataMember(Name = "physicalLocation", IsRequired = false, EmitDefaultValue = false)]
+        public PhysicalLocation PhysicalLocation { get; set; }
+
+        /// <summary>
+        /// The human-readable fully qualified name of the logical location. If run.logicalLocations is present, this value matches a property name within that object, from which further information about the logical location can be obtained.
+        /// </summary>
+        [DataMember(Name = "fullyQualifiedLogicalName", IsRequired = false, EmitDefaultValue = false)]
+        public string FullyQualifiedLogicalName { get; set; }
+
+        /// <summary>
+        /// The index within the logicalLocations array that specifies the logical location associated with the result.
+        /// </summary>
+        [DataMember(Name = "logicalLocationIndex", IsRequired = false, EmitDefaultValue = false)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [System.ComponentModel.DefaultValue(-1)]
```

Comment:
missing jschema functionality. we must manually apply these attributes.

Reply:
No, we have `AttributeHint` for that. See CodeGenHints.json for examples. In fact, you used it extensively to apply the `JsonConverter` attribute when you removed the contract resolver :-)

---
In reply to: [243699993](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243699993) [](ancestors = 243699993)

Reply:
Right. Still would be helpful to have more automatic code gen for this. Note that System.ComponentModel.DefaultValue is a standard attribute. A more important gap, mentioned in another comment thread: we need to initialize in ctor. I've updated comments in the issue below.

[https://github.com/Microsoft/jschema/issues/57](https://github.com/Microsoft/jschema/issues/57)

---
In reply to: [243719649](https://github.com/Microsoft/sarif-sdk/pull/1170#discussion_r243719649) [](ancestors = 243719649,243699993)